app/services/blacklight/search_service.rb

```diff
@@ -148,7 +148,11 @@ def fetch_many(ids, extra_controller_params)
               .merge(blacklight_config.fetch_many_document_params)
               .merge(extra_controller_params)
 
-      solr_response = repository.search(query)
+      solr_response = if repository.respond_to?(:find_all)
```

Comment:
Won't all repositories that extend from `AbstractRepository` now respond to find_all?  Should we add a deprecation warning if they don't implement `find_all`?

Reply:
That was my hope, but I don't think we can guarantee downstream applications have done that.

I'm not sure we need a deprecation warning until we want to remove the fallback to `search` but I don't see a pressing need for that 🤷‍♂️

Reply:
If not a deprecation, would you mind dropping a brief comment in the code here explaining why we would want to check first?